modules/statistics/php/charts.class.inc

```diff
@@ -0,0 +1,343 @@
+            $totalRecruitment = $DB->pselectOne(
+                "SELECT COUNT(c.CandID)
+                FROM candidate c
+                WHERE
+                c.RegistrationCenterID=:Site AND
+                c.Active='Y' AND
+                c.Entity_type='Human'",
+                array('Site' => $siteID)
+            );
+
+            $recruitmentBySiteData[] = array(
+                "label" => $siteName,
+                "total" => $totalRecruitment,
+            );
+        }
+        return new \LORIS\Http\Response\JsonResponse($recruitmentBySiteData);
+    }
+
+    /**
+     * Handle an incoming request for site sex breakdown.
+     *
+     * @return ResponseInterface
+     */
+    private function _handleSiteSexBreakdown()
+    {
+        $DB            = \NDB_Factory::singleton()->database();
+        $sexData       = array();
+        $list_of_sites = \Utility::getSiteList(true, false);
+
+        foreach ($list_of_sites as $siteID => $siteName) {
+            $sexData['labels'][] = $siteName;
+            $sexData['datasets']['female'][] = $DB->pselectOne(
+                "SELECT COUNT(c.CandID)
+                FROM candidate c
+                WHERE c.RegistrationCenterID=:Site
+                    AND c.Sex='female' AND c.Active='Y'
+                    AND c.Entity_type='Human'",
+                array('Site' => $siteID)
+            );
+            $sexData['datasets']['male'][]   = $DB->pselectOne(
+                "SELECT COUNT(c.CandID)
+                FROM candidate c
+                WHERE c.RegistrationCenterID=:Site AND c.Sex='male' AND c.Active='Y'
+                AND c.Entity_type='Human'",
+                array('Site' => $siteID)
+            );
+        }
+        return (new \LORIS\Http\Response\JsonResponse($sexData));
+    }
+
+    /**
+     * Handle an incoming request for monthly progression
+     *
+     * @return ResponseInterface
+     */
+    private function _handleScansByMonth()
+    {
+        $DB = \NDB_Factory::singleton()->database();
+
+        $scanData = array();
+        // Run a query to get all the data. Order matters to ensure that the
+        // labels are calculated in the correct order.
+        $data = $DB->pselect(
+            "SELECT s.CenterID,
+                CONCAT(MONTH(pf.Value), '-', YEAR(pf.Value)) as datelabel,
+                COUNT(distinct s.ID) as count
+            FROM files f
+            LEFT JOIN parameter_file pf USING (FileID)
+            LEFT JOIN session s ON (s.ID=f.SessionID)
+            JOIN parameter_type pt USING (ParameterTypeID)
+            WHERE pt.Name='acquisition_date'
+            GROUP BY MONTH(pf.Value), YEAR(pf.Value), s.CenterID
+            ORDER BY YEAR(pf.Value), MONTH(pf.Value), s.CenterID",
+            array()
+        );
+
+        // Create the labels
+        $labels = [];
+        foreach ($data as $row) {
+            $labels[$row['datelabel']] = true;
+        }
+        $scanData['labels'] = array_keys($labels);
```

Comment:
I'm not really clear on the use of $labels here and why it needs to be set to true. It doesn't seem to be used elsewhere.

It looks like this part could be simplified to

```php
foreach ($data as $row) {
    $scanData['labels'][] = $row['datalabel'];
}
```

or something even simpler.


Not a big deal just found it hard to parse.

Reply:
It doesn't need to be set to true, it just needs to be set to something so that it's an associative array in order to remove duplicate date labels from across sites. Your change would make it a numeric array and multiple sites with the same date label would appear multiple times.

Reply:
Fair enough. It's a small thing so we don't need to devote more brainpower to it